maker/src/lib.rs

```diff
@@ -50,6 +50,10 @@ pub struct Opts {
     #[clap(long)]
     pub verbose_spans: bool,
 
+    /// If enabled, browser UI is not automatically launched at startup.
+    #[clap(long)]
+    pub headless: bool,
```

Comment:
I propose to do it the otherway around. Pass in `--browser` to open the default browser. Our default platforms are servers and there it does not make sense to open a browser automatically.

Reply:
I went with this as this is the conventional flag name used for this purposes, even browsers themselves have a flag with thhis nome.  Moveover, on a server this will not open a browser anyway, so there's no need to specify the flag.

As far as I remember, the feature of opening the browser was meant for people running the binary and who had less experience with CLI.

Reply:
> As far as I remember, the feature of opening the browser was meant for people running the binary and who had less experience with CLI.

We never had anyone doing so and our efforts in https://github.com/itchysats/itchysats/issues/2889 would make this feature obsolete.

Reply:
I would also prefer it if we didn't open a browser window by default. Maybe just call the flag `disable_headless` or `auto_open_browser`. Even if the name isn't as intuitive.

Reply:
again, this is the standard terminology, you can see it in many programs (even likes of Chrome). There's even a wikipedia entry: https://en.wikipedia.org/wiki/Headless_software 

I don't see a reason why we should not enable GUI for casual users starting the binary. I have provided this feature for devs who start things multiple times, as I've heard that some of the devs would prefer to be able to turn it off. this PR or ticket are not about changing the default UX.

Reply:
> again, this is the standard terminology, you can see it in many programs (even likes of Chrome). There's even a wikipedia entry: https://en.wikipedia.org/wiki/Headless_software

I don't think it's a convincing argument to enable behaviour by default just because there is well-known terminology around it. The suggestion to rename the flag (and go against said terminology) is so that we don't have to enable the behaviour by default, not because I think the name is better.

> I don't see a reason why we should not enable GUI for casual users starting the binary. I have provided this feature for devs who start things multiple times, as I've heard that some of the devs would prefer to be able to turn it off. this PR or ticket are not about changing the default UX.

Sure, I can see why someone could benefit from this. We can keep the behaviour.

Reply:
I'm sorry, I misunderstood you; I thought we were just talking about the name.

The reason to keep the behaviour is mostly "principle of least surprise" - typically if you open something on desktop, then you want to start the UI if there is one.

Reply:
> The reason to keep the behaviour is mostly "principle of least surprise" - typically if you open something on desktop, then you want to start the UI if there is one.

I agree with this statement but we haven't seen any users using it on desktop.